Terminal.Gui/Application/MainLoop.cs

```diff
@@ -36,7 +37,7 @@ internal interface IMainLoopDriver
 ///     Monitoring of file descriptors is only available on Unix, there does not seem to be a way of supporting this
 ///     on Windows.
 /// </remarks>
-internal class MainLoop : IDisposable
+public class MainLoop : IDisposable
```

Comment:
Why does this need to be `public`?

Reply:
Again this is because it is a return on a member of IConsoleDriver

Reply:
I understand why @tznind needed to do that. When I was working on #3768 to leverage centralize common code I also needed to made `MainLoop` class to public because I needed to add it to the `ConsoleDriver`, which it's a public abstract class. Sometime it's the price to pay for to have centralized automatizations for all drivers.